src/evo/netinfo.h

```diff
@@ -60,6 +60,35 @@ constexpr std::string_view NISToString(const NetInfoStatus code)
     assert(false);
 }
 
+// Purpose corresponds to the index position in the ExtNetInfo map (which is ordered),
+// entries must be contiguous and cannot be changed once set without a format version
+// update, which will necessitate a hard-fork.
+namespace Purpose {
+enum : uint8_t {
+    // Mandatory for masternodes
+    CORE_P2P = 0,
+};
+} // namespace Purpose
+
+constexpr bool IsValidPurpose(const uint8_t purpose)
+{
+    switch (purpose) {
+    case Purpose::CORE_P2P:
+        return true;
+    } // no default case, so the compiler can warn about missing cases
+    return false;
+}
+
+// Warning: Used in RPC code, altering existing values is a breaking change
+constexpr std::string_view PurposeToString(const uint8_t purpose, const bool lower = false)
```

Comment:
here should be enum instead uint8_t

```
constexpr std::string_view PurposeToString(const Purpose purpose, const bool lower = false)

Reply:
it seems as capitals are used only in logs. If it is true, consider to drop an argument `lower=false` and just always output it (in RPC and in logs) as lowercase without capital option. It doesn't seems much useful for me

Reply:
Used lowercase and dropped `lower` argument, resolved in latest push.